source/class/qx/tool/compiler/app/Cldr.js

```diff
@@ -59,9 +60,24 @@ qx.Class.define("qx.tool.compiler.app.Cldr", {
       }
       log.debug("Loading CLDR " + cldrPath);
 
-      return readFile(path.join(cldrPath, data_path, locale + ".xml"), {
-        encoding: "utf-8"
+      const fullDir = path.join(cldrPath, data_path);
+
+      return readDir(fullDir).then(names => {
+        return new Promise((resolve, reject) => {
+          const searchedName = locale.toLowerCase() + ".xml";
+          const realName = names.find(name => name.toLowerCase() === searchedName);
```

Comment:
This looks pretty liberal, allowing any mixed case. Maybe that's ok, but might it be more appropriate to check for the two known options (fully uppercase, or capitalized)?

Reply:
@derrell How am I supposed to check these cases: ff_Latn_SN.xml, be_TARASK.xml, en_US_POSIX.xml?
I mean parse string for 2 or 3 parts and make on them some transformations (into full or capitalized). In examples the second part could be or fully uppercase or capitalized. The checking code would be difficult. Does it worth it? There are around 600 locales.

Reply:
Do we want to allow any case? Is there a specification that dictates what the case should be? I think you'd mentioned previously that there were at least two valid options of case. It's fine to be liberal as long as that isn't going to cause any problems. Is there ever a time where two different mixes of case of the same text are *supposed* to be different? If not, then no, it's likely not worth it at all.

Reply:
I'll approve on the assumption that there are no such times. If you can think of any, we can discuss how to best implement it.

Reply:
@derrell I was wrong when said there are two cases. Actually there are more than 2 choices. You may see all file names [here](https://github.com/papandreou/node-cldr/tree/master/3rdparty/cldr/common/main). At the moment in Windows if you set up for example a locale name in 'compile.json' like sR_laTin it will work.